handlers/sf-billing-account-remover/src/main/scala/com/gu/sf_billing_account_remover/BillingAccountRemover.scala

```
@@ -0,0 +1,312 @@
+    zuoraConfig: ZuoraConfig,
+    recordsForUpdate: Seq[BillingAccountsRecords.Records]
+  ): Seq[BillingAccountsRecords.Records] = {
+
+    for {
+      billingAccount <- recordsForUpdate
+      zuoraCancellationResponses <- updateBillingAccountInZuora(
+        zuoraConfig,
+        billingAccount
+      )
+    } yield zuoraCancellationResponses
+
+  }
+
+  def updateBillingAccountInZuora(
+    zuoraConfig: ZuoraConfig,
+    accountToDelete: BillingAccountsRecords.Records
+  ): Option[BillingAccountsRecords.Records] = {
+
+    val response =
+      updateZuoraBillingAcc(
+        zuoraConfig,
+        BillingAccountsForRemoval().asJson.spaces2,
+        accountToDelete.Zuora__External_Id__c
+      )
+
+    val parsedResponse = decode[ZuoraResponse](response)
+
+    parsedResponse match {
+      case Right(dr) =>
+        if (dr.Success) {
+          None
+        } else {
+          Some(
+            accountToDelete.copy(
+              ErrorCode = Some(dr.Errors(0).Code),
+              ErrorMessage = Some(dr.Errors(0).Message)
+            )
+          )
+        }
+      case Left(ex) => {
+        Some(accountToDelete.copy(ErrorMessage = Some(ex.toString)))
+      }
+    }
+
+  }
+
+  def updateZuoraBillingAcc(zuoraConfig: ZuoraConfig,
+                            billingAccountForRemovalAsJson: String,
+                            zuoraBillingAccountId: String) = {
+    Http(
+      s"${zuoraConfig.zuoraInstanceUrl}/v1/object/account/$zuoraBillingAccountId"
+    ).header("apiAccessKeyId", zuoraConfig.apiAccessKeyId)
+      .header("apiSecretAccessKey", zuoraConfig.apiSecretAccessKey)
+      .header("Content-Type", "application/json")
+      .option(HttpOptions.readTimeout(30000))
+      .put(billingAccountForRemovalAsJson)
+      .asString
+      .body
+  }
+
+  def writeErrorsBackToSf(
+    sfAuthDetails: SfAuthDetails,
+    failedUpdates: Seq[BillingAccountsRecords.Records]
+  ): Unit = {
+
+    updateBillingAccountsInSf(sfAuthDetails, failedUpdates)
```

Quick way to avoid silent error would be

```
(for {
  _ <- updateBillingAccountsInSf(sfAuthDetails, failedUpdates)
  _ <- insertErrorRecordsInSf(sfAuthDetails, failedUpdates)
} yield ()).left.foreach(e => throw e)
```

---

Great, this is done